apps/web/src/app/challenge/_components/comments/enhanced-user-badge.getProfileData.ts

```diff
@@ -0,0 +1,45 @@
+'use server';
+
+import { prisma } from '@repo/db';
+import { RoleTypes, type Role } from '@repo/db/types';
+import { getBadges } from '~/app/(profile)/[username]/_components/dashboard/_actions';
+
+export async function getProfileData(username: string) {
+  const user = await prisma.user.findFirst({
```

Comment:
`findFirstOrThrow` then you can avoid the manual throw after this

Reply:
🤘🏽🤘🏽